components/usage/pkg/controller/reconciler.go

```diff
@@ -247,7 +248,7 @@ func usageReportToUsageRecords(report UsageReport) []db.WorkspaceInstanceUsage {
 				AttributionID: attributionId,
 				StartedAt:     instance.CreationTime.Time(),
 				StoppedAt:     stoppedAt,
-				CreditsUsed:   0,
+				CreditsUsed:   float64(pricer.CreditsUsedByInstance(&instance, now)),
```

Comment:
about the float64: I faintly recall we report credits as ints to stripe, right? If yes, we probably should do the same here, and adjust the DB/gRPC model if they don't agree.

@andrew-farries WDYT?

Reply:
The designs here [internal](https://gitpod.slack.com/archives/C027AUU7BC3/p1657709179313939) show credits displayed with fractions.

But I think you are right in that we should be consistent in allowing the credits per minute for a workspace class be fractional but always bill and display whole numbers of credits.

I'll update the types in a follow-up.

Reply:
My understanding was that we have to report total usage to Stripe as an int, but we should collect usage per session in fractional units to avoid rounding errors.

Reply:
As explained [here](https://github.com/gitpod-io/gitpod/pull/11429#issuecomment-1187105119), @jldec is right in thinking that this should in fact remain a float.